pybamm/simulation.py

```diff
@@ -185,6 +187,8 @@ def set_up_and_parameterise_experiment(self):
         This needs to be done here and not in the Experiment class because the nominal
         cell capacity (from the parameters) is used to convert C-rate to current.
         """
+        msg = "pybamm.simulation.set_up_and_parameterise_experiment is not meant to be accessed directly."
+        warn(msg, DeprecationWarning)
```

Comment:
Given that not a lot of users are using these methods and we want to make them private in the next release - we could create dummy public methods that throw `NameError` on being invoked -
```python
def set_up_and_parameterise_experiment(self):
    """
    A note here for users that this should not be used and is not private
    with the name _set_up_and_parameterise_experiment
    """
    raise NameError(...)
```

Just suggestions. We should go ahead with the strategy that most maintainers agree with.

Reply:
Should I implement it or wait for the other maintainers feedback on this approach??

Reply:
I don't think it's a good idea to make them private right in the next release, since it doesn't conform to our policy of three releases – even if the popularity of this API not being relatively high is the question. Keeping them public plus raising a DeprecationWarning should be enough, a NameError sounds a bit odd IMO (I don't think we have used this before).